backend/src/user_commands.py

```diff
@@ -2,165 +2,517 @@
+    sell_cmd = Database.find_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user'], 'sell': { '$exists': True } }, { 'sell': 1, '_id': 0})
+
+    # Not sure if to include this because it would be cancelled either way
+    # Check that less than 60s has passed
+    # sec_passed = time.time() - float(buy_cmd['sell']['timestamp'])
+    # if (sec_passed <= 60):
+
+    Database.update_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user']}, {'$unset': { 'sell': ""}})
+
     log.logEvents['userCommand'](cmdDict)
 
+    cmdCompleted(cmdDict)
+
+
 def CMD_SetBuyAmt(cmdDict, threadContext):
+    """
+        Creates a buy trigger based on the number of the stocks the user wants to buy
+    """
     printCmd(cmdDict)
 
-    try:
+    # Check user has enough funds in account
+    user_funds = Database.find_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user']}, { 'funds': 1, '_id': 0})
+    if (user_funds['funds'] >= float(cmdDict['amount'])):
+        # Add new trigger buy command and remove funds from user account
+        buy_trigger = {
+            'user': cmdDict['user'],
+            'type': 'buy',
+            'stockSymbol': cmdDict['stockSymbol'],
+            'amount': cmdDict['amount']
+        }
+        Database.insert(TRIGGER_COLLECT, buy_trigger)
+        Database.update_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user']}, {'$inc': {'funds': -cmdDict['amount']}})
+
         cmdDict['timestamp'] = str(int(time.time()))
+        cmdDict['amount'] = str(int(cmdDict['amount']))
         log.logEvents['userCommand'](cmdDict)
-    except:
+    else:
         cmdDict['timestamp'] = str(int(time.time()))
+        cmdDict['amount'] = str(int(cmdDict['amount']))
         cmdDict['errorMessage'] = "Insufficient funds. Cannot set buy amount." 
 
         log.logEvents['errorEvent'](cmdDict)
+    
+    cmdCompleted(cmdDict)
 
 def CMD_CancelSetBuy(cmdDict, threadContext):
+    """
+        Cancels the set buy command
+    """
     printCmd(cmdDict)
 
+    # Check if person has already set buy amount for the stock
+    buy_trigger = Database.find_one(TRIGGER_COLLECT, {'user': cmdDict['user'], 'stockSymbol': cmdDict['stockSymbol'], 'type': 'buy'})
+    if  (buy_trigger != None):
+        # Add funds back to user's account and delete set_buy trigger
+        Database.update_one(ACCOUNTS_COLLECT, {'_id': cmdDict['user']}, {'$inc': {'funds': buy_trigger ['amount']}})
+        Database.remove(TRIGGER_COLLECT, {'user': cmdDict['user'], 'stockSymbol': cmdDict['stockSymbol'], 'type': 'buy'})
+    
     cmdDict['timestamp'] = str(int(time.time()))
     log.logEvents['userCommand'](cmdDict)
 
+    cmdCompleted(cmdDict)
+    
 def CMD_SetBuyTrigger(cmdDict, threadContext):
+    """
+        Adds the price trigger to the set buy command
+    """
+
     printCmd(cmdDict)
 
+    # Check if person has already set buy amount for the stock
+    if  (Database.find_one(TRIGGER_COLLECT, {'user': cmdDict['user'], 'stockSymbol': cmdDict['stockSymbol'], 'type': 'buy'}) != None):
```

Comment:
maybe a separate transaction server can handle these triggers aswell as query the stock server every 10ms?

Reply:
I think that's what Dianna and I were talking about on Thursday. It's a great idea, will do!